BasicConcepts.ipynb

```diff
@@ -0,0 +1,469 @@
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "1 2 3 4 5 6 7 8\n"
+     ]
+    }
+   ],
+   "source": [
+    "age = 1\n",
+    "Age = 2\n",
+    "aGe = 3\n",
+    "AGE = 4\n",
+    "a_g_e = 5\n",
+    "_age = 6\n",
+    "age_ = 7\n",
+    "_AGE_ = 8\n",
+    "\n",
+    "print(age, Age, aGe, AGE, a_g_e, _age, age_, _AGE_)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "id": "4b1c5550-e68e-41fb-b8de-41a795b3542a",
+   "metadata": {},
+   "outputs": [
+    {
+     "ename": "SyntaxError",
+     "evalue": "cannot assign to operator (1315807947.py, line 1)",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[0;36m  Cell \u001b[0;32mIn[16], line 1\u001b[0;36m\u001b[0m\n\u001b[0;31m    my-var = \"Maria\"\u001b[0m\n\u001b[0m    ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m cannot assign to operator\n"
+     ]
+    }
+   ],
+   "source": [
+    "my-var = \"Maria\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 40,
+   "id": "268fb9e4-0e70-4019-8791-3066d69ec06f",
+   "metadata": {},
+   "outputs": [
+    {
+     "ename": "SyntaxError",
+     "evalue": "invalid syntax (3650524433.py, line 1)",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[0;36m  Cell \u001b[0;32mIn[40], line 1\u001b[0;36m\u001b[0m\n\u001b[0;31m    1myvar = \"Fabian\"\u001b[0m\n\u001b[0m     ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
+     ]
+    }
+   ],
+   "source": [
+    "1myvar = \"Fabian\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "id": "07ab67c8-2561-4e98-8b67-92ae5ff255d4",
+   "metadata": {},
+   "outputs": [
+    {
+     "ename": "SyntaxError",
+     "evalue": "invalid syntax (191355952.py, line 1)",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[0;36m  Cell \u001b[0;32mIn[17], line 1\u001b[0;36m\u001b[0m\n\u001b[0;31m    for = 3\u001b[0m\n\u001b[0m        ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
+     ]
+    }
+   ],
+   "source": [
+    "for = 3"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a5d7171c-4087-425a-9f93-3daaf88fbbd9",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "### Built-in Data Types\n",
+    "\n",
+    "Python has the following categories and data types built-in by default:\n",
+    "\n",
+    "* Boolean Type: bool\n",
+    "* Text Type: str\n",
+    "* Numeric Types: int, float, complex\n",
+    "* Sequence Types: list, tuple, range\n",
+    "* Mapping Type: dict\n",
+    "* Set Types: set, frozenset\n",
+    "* Binary Types: bytes, bytearray, memoryview\n",
+    "* None Type: NoneType\n",
+    "\n",
+    "Notes:\n",
+    "- The built-in Python type() function can be used to check the variable type\n",
+    "- It can be assigned a value of one type and then later re-assigned a value of a different type."
```

Comment:
Even though a variable was created with certain type, it can be re-assigned with another value that has a different type.